app/src/main/java/org/dhis2/usescases/datasets/datasetInitial/DataSetInitialActivity.java

```diff
@@ -94,11 +95,11 @@ public void setData(DataSetInitialModel dataSetInitialModel) {
         checkActionVisivbility();
     }
 
-    private void clearCatOptionCombo(){
-        if(!binding.getDataSetModel().categoryComboName().equals("default")){
-            for(int i=0; i<binding.catComboContainer.getChildCount();i++){
+    private void clearCatOptionCombo() {
+        if (!binding.getDataSetModel().categoryComboName().equals("default")) {
+            for (int i = 0; i < binding.catComboContainer.getChildCount(); i++) {
                 View catView = binding.catComboContainer.getChildAt(i);
```

Comment:
what is catComboContainer?

Reply:
Is the layout containing each category view

Reply:
would it crashes if the XML changes?

Reply:
Don't think so. It had to change a lot for it to fail.
I would not care too much about it. The idea in the future is getting rid of this activity and add all its logic to a fake first section in the event data entry form.